src/wp-includes/https-detection.php

```diff
@@ -68,16 +68,9 @@ function wp_is_site_url_using_https() {
  * @return bool True if HTTPS is supported, false otherwise.
  */
 function wp_is_https_supported() {
-	$https_detection_errors = get_option( 'https_detection_errors' );
+	$https_detection_errors = wp_get_https_detection_errors();
```

Comment:
This function is quite expensive as it makes HTTP requests. This means `wp_is_https_supported()` is equally expensive, since it no longer relies on a stored / cached value.

I think we should clarify this in the documentation of the function. Maybe add something like: `This function performs HTTP requests and therefore should be used with caution.`

We should also add the same comment to `wp_get_https_detection_errors()` IMO.

Last but not least, the doc-block of `wp_get_https_detection_errors()` still mentions something about "a regular Cron hook", which is no longer correct as of https://core.trac.wordpress.org/ticket/58494 / https://core.trac.wordpress.org/changeset/56664. It also misses the `@return` annotation, even though it returns an array of potential errors.

cc @adamsilverstein @johnbillion @audrasjb